### ISSUE TITLE
Interface for CO3D

### ISSUE THREAD

**Issue by TLB-MISS** [DarlingHang/radiance_field_propagation#1]
Hi.
First of all, thank you for your wonderful work.
When will the works in TODO be updated, especially "Interface for CO3D"?

Thank you in advance.

**Comment by TLB-MISS** [comment]
Additional Question

I tried below code(in the [README](https://github.com/DarlingHang/radiance_field_propagation)) with default setting
`python train.py --config configs/flower.txt`
But results are quite bad.

Below 4 images are outputs of last iteration.
They are no improvement at all.
Are the provided configuration settings not same with the configuration in the thesis experiment?
![009999_001_mask](https://user-images.githubusercontent.com/77427895/217274186-67db686d-91e1-4d8a-8eea-95814770af27.png)
![009999_002_mask](https://user-images.githubusercontent.com/77427895/217274192-de448dc1-d8bb-44cb-8789-6fd0281d49dc.png)
![009999_003_mask](https://user-images.githubusercontent.com/77427895/217274195-fa6e0e92-875d-47c3-abc3-c542b749ff59.png)
![009999_004_mask](https://user-images.githubusercontent.com/77427895/217274197-56e0fe81-5081-486e-aea9-94d0fce26d19.png)

**Comment by DarlingHang** [comment]
Thanks for reaching out. Your results are reasonable from my side and basically the same as those from my trial. To improve the results, you may need better initial labels. One way of getting them is to run IEM with customized parameters.

**Comment by DarlingHang** [comment]
I will finish the interface asap, hopefully after my current projects. One way to bypass this is to run colmap by yourself on CO3D scenes and use the provided interfaces.

**Comment by TLB-MISS** [comment]
Thanks for quick reply!

As far as I know, CO3D converts colmap results into Quaternions and provides them.
However, regardless of camera parameters, I am curious about setting magic numbers such as self.near_far and self.scene_bbox.

**Comment by TLB-MISS** [comment]
> Thanks for reaching out. Your results are reasonable from my side and basically the same as those from my trial. To improve the results, you may need better initial labels. One way of getting them is to run IEM with customized parameters.

And aren't the masks you provide [here](https://drive.google.com/drive/folders/1kjRf-PSoP0uGWYql_Ty3TFjFRHt6R32W) the initial ones from IEM?

**Comment by DarlingHang** [comment]
For CO3D scenes, I remember we used scene_bbox [[-10, -10, -10], [10, 10, 10]]. Not 100% sure, but you can adjust it based on your observation. From our experiments, it is not easy to achieve the sound novel view synthesis on CO3D dataset, even with another model (instant-ngp, other nerf implementations).

**Comment by DarlingHang** [comment]
I got better masks than what I provided, using more customized parameters. You can check the IEM paper and repo for more insights. If you are looking for perfect masks for flower and fortress, you can find them [here](https://drive.google.com/file/d/1gD5paJ8HBOFVyMRgweTc0jNTL41SmBFz/view).

**Comment by TLB-MISS** [comment]
Thank you for sharing!

**Comment by TLB-MISS** [comment]
@DarlingHang 

> I will finish the interface asap, hopefully after my current projects. One way to bypass this is to run colmap by yourself on CO3D scenes and use the provided interfaces.

Sorry for the rush, but when will the CO3D interface be implemented?
If it takes some time, could you check out my code(CO3D with colmap) instead?
I used given CO3D masks as initial masks. But the output segmentation is weird.

I first got `poses_bounds.npy` from the colmap output by referring to the code [here](https://github.com/Fyusion/LLFF/tree/master/llff/poses).
Next, I set the three variables below.
```
self.near_far = [0.0, 1.0]
self.scene_bbox = torch.tensor([[-8., -8., -8.], [8., 8., 8.]])
scale_factor = near_original * 9
```

**Output Example**
![008999_001_mask_gt](https://user-images.githubusercontent.com/77427895/225321857-c5bd7bce-4f8a-41dc-a779-d1d69bb557df.png)
![008999_005_mask_gt](https://user-images.githubusercontent.com/77427895/225321974-5c27888c-0813-485e-a852-89efe3294ea3.png)

I'm not sure if my implementation is wrong, or if my magic numbers (`self.scene_bbox`, `self.near_far`, `scale_factor`, etc) settings are wrong.
If you need, I'll give my code and mini dataset to you.

Thanks.

**Comment by DarlingHang** [comment]
You can verify whether there's something wrong with bbox/near_far by checking the rendered rgb image. In most cases, colmap poses are not good enough for CO3D to give photo-realistic rendering.

Also, this object is too textured and I guess our method wouldn't work well on it. Try to use something as easy as LLFF flowers.

**Comment by buaadf** [comment]
> 

So if I want to get better masks, I have to provide better initial masks?
